README.md

```diff
@@ -21,36 +33,98 @@ If you login to a project, you will not have to provide the `--key` option becau
 rdme login
 ```
 
-### Uploading a new Swagger file to ReadMe
+## Usage
+### `rdme` flags
+```
+swaggerfile.json || oasfile.yaml
+--key              # API key associated with your ReadMe project
+--version          # The version
+--id               # The id of a OAS file previously uploaded. This is available through uploading a new file through this CLI
+--fork             # The semantic version which you'd like to fork from
+--codename         # The codename or nickname for a particular version
+--main             # Should this version be the primary (default) version for your project?
+--beta             # Is this version in beta?
+--isPublic         # Would you like to make this version public? Any primary version must be public
+```
 
+### Swagger
+#### Uploading a new Swagger file to ReadMe
+This will return an id and url for you to update your file and view it in the client, respectively
 ```sh
 rdme swagger {path-to-swagger.json} --key={api-key}
 ```
 
-### Editing an existing Swagger file
+#### Editing an existing Swagger file
 ```sh
 rdme swagger {path-to-swagger.json} --key={api-key} --id={existing-id}
 ```
 
-### Syncing a folder of markdown docs to ReadMe
+#### Uploading or editing a swagger file including with specified version
+You can additional include a version flag, specifying the target version for your file's destination
+```sh
+rdme swagger {path-to-swagger.json} --key={api-key} --version={project-version}
+```
+
+#### Omitting the file path
+If you run `rdme` within a directory that contains your Swagger or OAS file, you can omit the file path.
+Be sure to use one of the following file names: `swagger.json`, `swagger.yaml`, `openapi.json`, `openapi.yaml` if you do.
+```sh
+rdme swagger --key={api-key}
+```
+
+### Docs
+#### Syncing a folder of markdown docs to ReadMe
 
 ```sh
 rdme docs path-to-markdown-files --key={api-key} --version={project-version}
 ```
 
-### Edit a single readme doc on your local machine
+#### Edit a single readme doc on your local machine
 
 ```sh
 rdme docs:edit <slug> --key={api-key} --version={project-version}
 ```
 
-### Open your ReadMe project in your browser
+### Versions
+#### Get all versions associated with your project
+```sh
+rdme versions --key={api-key}
+```
 
+#### Get all information about a particular version
+```sh
+rdme versions:versionId --key={api-key} --version={project-version}
+```
+
+#### Create a new version using flags
+```sh
+rdme versions:create --key={api-key} --version={project-version} --fork={version-fork} --codename={version-name} --main --beta
+```
+
+#### Create a new version without flags
+Creating a version without version-specific flags will allow the ReadMe CLI to prompt you with configuration options
+```sh
+rdme versions:create --key={api-key} --version={project-version}
+```
+
+#### Update a version
+The command to update a version takes the same flags as creating a new version
+```sh
+rdme versions:update --key={api-key} --version={project-version}
```

Comment:
I really like how well this is documented. Can we also add the parameter specifications in the `rdme --help` command as well?

Reply:
@erunion is discussing some beautification to CLI output in general. This seems like a good candidate to bundle there.

Reply:
Yep! I'm working on some QOL improvements to the CLI in general to allow for usage descriptions on every command.